client/components/material-icon/index.jsx

```diff
@@ -0,0 +1,42 @@
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+import React from 'react';
+import PropTypes from 'prop-types';
+import spritePath from '@automattic/material-design-icons/svg-sprite/material-icons.svg';
```

Comment:
@griffbrad This line is interesting. How does this import get converted into a path string, and how would I be able to replicate it in gridicons?

Reply:
https://github.com/Automattic/wp-calypso/blob/master/packages/calypso-build/webpack/file-loader.js

file-loader magic

Reply:
This is done with webpack's file loader.  In the context of Calypso, it's added to webpack.config.js here:
https://github.com/Automattic/wp-calypso/blob/master/webpack.config.js#L234

That comes out of the Calypso build package:
https://github.com/Automattic/wp-calypso/blob/master/packages/calypso-build/webpack/file-loader.js

Reply:
Very interesting, thank you both! Should be easy to replicate for gridicons, then :)